src/main/java/core/basesyntax/Figure.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax;
+
+public abstract class Figure extends FigureSupplier implements CollorSupplier, GetArray {
+    private String color;
+
+    public void setColor() {
+        this.color = getRandomColor();
+    }
+
+    public String getColor() {
+        return this.color;
+    }
+
+    public abstract double getArea();
+
+    public abstract void draw();

```

Comment:
lets create some InformationDrawer and AreaCalculator interfaces with getArea() and draw() abstract methods and implement them in Figure class